modules/infinite-scroll/themes/twentytwelve.php

```diff
@@ -8,41 +8,36 @@
 /**
  * Add theme support for infinite scroll
  */
-function twenty_twelve_infinite_scroll_init() {
+function jetpack_twentytwelve_infinite_scroll_init() {
 	add_theme_support( 'infinite-scroll', array(
 		'container'      => 'content',
-		'footer'         => 'page'
+		'footer'         => 'page',
+		'footer_widgets' => jetpack_twentytwelve_has_footer_widgets(),
 	) );
 }
-add_action( 'after_setup_theme', 'twenty_twelve_infinite_scroll_init' );
+add_action( 'after_setup_theme', 'jetpack_twentytwelve_infinite_scroll_init' );
 
 /**
  * Enqueue CSS stylesheet with theme styles for infinity.
  */
-function twenty_twelve_infinite_scroll_enqueue_styles() {
+function jetpack_twentytwelve_infinite_scroll_enqueue_styles() {
 	if ( wp_script_is( 'the-neverending-homepage' ) ) {
 		// Add theme specific styles.
 		wp_enqueue_style( 'infinity-twentytwelve', plugins_url( 'twentytwelve.css', __FILE__ ), array( 'the-neverending-homepage' ), '20120817' );
 	}
 }
-add_action( 'wp_enqueue_scripts', 'twenty_twelve_infinite_scroll_enqueue_styles', 25 );
+add_action( 'wp_enqueue_scripts', 'jetpack_twentytwelve_infinite_scroll_enqueue_styles', 25 );
 
 /**
- * Handle `footer_widgets` argument for mobile devices
- *
- * @param bool $has_widgets
- * @uses jetpack_is_mobile, is_front_page, is_active_sidebar
- * @filter infinite_scroll_has_footer_widgets
- * @return bool
+ * Do we have footer widgets?
  */
-function twenty_twelve_has_footer_widgets( $has_widgets ) {
+function jetpack_twentytwelve_has_footer_widgets() {
 	if ( function_exists( 'jetpack_is_mobile' ) && jetpack_is_mobile() ) {
 		if ( is_front_page() && ( is_active_sidebar( 'sidebar-2' ) || is_active_sidebar( 'sidebar-3' ) ) )
-			$has_widgets = true;
+			return true;
 		elseif ( is_active_sidebar( 'sidebar-1' ) )
-			$has_widgets = true;
+			return true;
 	}
 
-	return $has_widgets;
+	return false;
 }
-add_filter( 'infinite_scroll_has_footer_widgets', 'twenty_twelve_has_footer_widgets' );
```

Comment:
Hey @georgestephanis, I have removed the filters, because I have added `footer_widgets` to `add_theme_support` arguments list. 
I have tested all themes and everything was working the same way like it was with filter.